include/exadg/structure/material/library/st_venant_kirchhoff.h

```diff
@@ -69,16 +69,28 @@ class StVenantKirchhoff : public Material<dim, Number>
                     StVenantKirchhoffData<dim> const &      data);
 
   dealii::Tensor<2, dim, dealii::VectorizedArray<Number>>
-  evaluate_stress(dealii::Tensor<2, dim, dealii::VectorizedArray<Number>> const & E,
-                  unsigned int const                                              cell,
-                  unsigned int const                                              q) const final;
+  second_piola_kirchhoff_stress(
+    dealii::Tensor<2, dim, dealii::VectorizedArray<Number>> const & strain_measure,
+    unsigned int const                                              cell,
+    unsigned int const                                              q) const final;
 
   dealii::Tensor<2, dim, dealii::VectorizedArray<Number>>
-  apply_C(dealii::Tensor<2, dim, dealii::VectorizedArray<Number>> const & E,
-          unsigned int const                                              cell,
-          unsigned int const                                              q) const final;
+  second_piola_kirchhoff_stress_derivative(
+    dealii::Tensor<2, dim, dealii::VectorizedArray<Number>> const & Grad_delta,
+    dealii::Tensor<2, dim, dealii::VectorizedArray<Number>> const & F_lin,
+    unsigned int const                                              cell,
+    unsigned int const                                              q) const final;
 
 private:
+  /*
+   * The second Piola-Kirchhoff stress tensor S is given as S = lambda * I * tr(E) + 2 mu E,
+   * with E being the Green-Lagrange strain tensor and Lamee parameters lambda and mu.
+   * Factor out coefficients for faster computation
+   * Sii = f0 * Eii + f1 * Eij, for i, j = 1, ..., dim and i != j, and
+   * Sij = f2 * (Eij + Eji),    for i, j = 1, ..., dim and i != j.
```

Comment:
this also needs to be adjusted according to https://github.com/exadg/exadg/pull/537/files#r1284082874